hare3/hare.go

```diff
@@ -339,8 +339,11 @@ func (h *Hare) Handler(ctx context.Context, peer p2p.Peer, buf []byte) error {
 			zap.Uint32("lid", msg.Layer.Uint32()),
 			zap.Stringer("sender", equivocation.Messages[0].SmesherID))
 		proof := equivocation.ToMalfeasanceProof()
-		if err := identities.SetMalicious(
-			h.db, equivocation.Messages[0].SmesherID, codec.MustEncode(proof), time.Now()); err != nil {
+		err := h.db.WithTx(context.Background(), func(tx sql.Transaction) error {
+			return identities.SetMalicious(
+				h.db, equivocation.Messages[0].SmesherID, codec.MustEncode(proof), time.Now())
+		})
+		if err != nil {
```

Comment:
Could you please explain why a transaction is used? Btw, the `tx` isn't used as `h.db` is passed to `identities.SetMalicious()` - is it on purpose?

Reply:
SetMalicious runs more than one statement, if they are executed without transaction then on failure the state will be inconsistent, e.g only one of the atxs will be marked as malicious